SAP/spartacus
# feat: CXSPA-6576 Introduce SERVER_ERROR_RESPONSE_FACTORY

This PR introduces the SERVER_ERROR_RESPONSE_FACTORY strategy to `ServerRespondingErrorHandler` to generate proper CxServerError errors depending on the incoming HTTP error response during SSR. Two default factories have been introduced together with this PR :
- CmsPageNotFoundServerErrorResponseFactory - responsible for generating errors when the CMS page is not found
- UnknownServerErrorResponseFactory- fallback, generates unknown server error for everything else


QA steps:

- call HTTP error that may be caught during server-side rendering e.g use HTTP proxy OR 
  - modify `pages` property in `default-cms-config.ts` to generate `CmsPageNotFoundServerError`
  - modify `components` property in `default-cms-config.ts` to generate `UnknownServerError`
- add `console.log` to `handleError` method inside core-libs/setup/ssr/error-handling/multi-error-handlers/server-responding-error-handler.ts
- run npm `run dev:ssr`
- open the application in the browser
- verify whether the appropriate error is visible in the terminal where server runs

closes [CXSPA-6575](https://jira.tools.sap/browse/CXSPA-6576)
